src/tests/unit/tests/injected/client-store-listener.test.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { IMock, It, Mock, MockBehavior, Times } from 'typemoq';
+
+import { BaseClientStoresHub } from '../../../../common/stores/base-client-stores-hub';
+import { ClientStoreListener, TargetPageStoreData } from '../../../../injected/client-store-listener';
+
+describe('ClientStoreListener', () => {
+    describe('initialize', () => {
+        let testSubject: ClientStoreListener;
+        let storeHubMock: IMock<BaseClientStoresHub<TargetPageStoreData>>;
+        let onReadyToExecuteVisualizationUpdatesMock: IMock<(storeData: TargetPageStoreData) => void>;
+        let triggerOnChange: () => void;
+
+        beforeEach(() => {
+            storeHubMock = Mock.ofType<BaseClientStoresHub<TargetPageStoreData>>(null, MockBehavior.Strict);
+            onReadyToExecuteVisualizationUpdatesMock = Mock.ofType<(storeData: TargetPageStoreData) => void>();
+
+            storeHubMock
+                .setup(shm => shm.addChangedListenerToAllStores(It.isAny()))
+                .callback(givenCallback => (triggerOnChange = givenCallback));
```

Comment:
looks like `triggerOnChange` is been called on each tests... does it make sense to call it right here on the setup? like this:
```typescript
  storeHubMock
                .setup(shm => shm.addChangedListenerToAllStores(It.isAny()))
                .callback(givenCallback => givenCallback());
```

Reply:
Good question! Since the setup for the interactions with the storeHubMock are different for each test/scenario, and as such in the test blocks, the function can only be triggered after that setup is complete.

Reply:
It will be trigger only when you actuate on the test subject though, right?